### PR TITLE
Update dependencies and target frameworks and fix build warnings

### PR DESCRIPTION
**Pull request by mus65**
Starting with .NET 9, NuGet Audit will transitively scan for CVEs, so consuming applications will get a warning about System.Text.Json:

>  warning NU1903: Package 'System.Text.Json' 6.0.1 has a known high severity vulnerability, https://github.com/advisories/GHSA-8g4q-xg66-9fp4

#24 already optimized this to only affect .NET Framework (but it's still not released).

This updates all dependencies (except Nunit 4 which has a lot of breaking changes) and cleans up some build warnings:

- removed netstandard2.0 from test projects. This doesn't make sense because you can't run tests against netstandard2.0 (LangVersion also isn't needed anymore then)
- removed EOL net7.0 target and added net8.0
- removed `Microsoft.CodeAnalysis.NetAnalyzers`because it's part of the SDK since .NET 5

### CODE REVIEW

**Comment by xfischer** [comment]
Hi ! Thanks a lot for your contribution. Will review and ask a review from @janusw (I have to add it here also) who did so much lately from the main repo. 
Please allow a couple of hours as I don't have access to my laptop now.

**Review by janusw — COMMENTED** [review]
I don't see anything wrong with this, except that probably the GHA hooks need to be updated to deal with .NET 8 as well. Maybe someone can approve the workflows to run, so that we can see them fail? 😆

**Comment by mus65** [comment]
Good point, I also updated the GitHub actions and dropped the EOL .NET versions from CI.

**Comment by janusw** [comment]
> Good point, I also updated the GitHub actions and dropped the EOL .NET versions from CI.

👍 Now we only need someone to approve the workflows so that they will actually run. @xfischer Can you do that?

**Review by xfischer — APPROVED** [review]

**Comment by janusw** [comment]
Alright, so the build job runs fine, but the benchmarking job fails. That one also needs to be adjusted:
https://github.com/GeoJSON-Net/GeoJSON.Text/blob/main/.github/workflows/continous-benchmark.yml

**Comment by xfischer** [comment]
@janusw @mus65 thanks to both of you.
@janusw I've added you as a maintainer

**Comment by mus65** [comment]
Benchmarks should hopefully be fixed now, but I guess the workflow needs another approval. :smile: I removed net6.0 from the benchmark project since .NET 6 is EOL next month anyway.

**Comment by janusw** [comment]
> Benchmarks should hopefully be fixed now, but I guess the workflow needs another approval. 😄

Since I'm a maintainer, I can do that now. 💪 (thanks, @xfischer !)

**Comment by janusw** [comment]
Alright, all the CI jobs run fine by now and the PR was approved already, so I'll just merge it ...

**Pull request merged**
